src/development/variables.md

```diff
@@ -1,17 +1,35 @@
 # Variables
 
-Platform.sh allows a high degree of control over both the build process and the runtime environment of a project.  Part of that control comes in the form of *variables* that are set independently of the project's code base but available either at build or runtime for your code to leverage.  Platform.sh also exposes additional information to your application that way, including information like database credentials, the host or port it can use, and so forth.
+<!-- toc -->
 
-## Types of variables
+Platform.sh allows a high degree of control over both the build process and the runtime environment of a project.  Part of that control comes in the form of *variables* that are set independently of the project's code base but available either at build or runtime for your code to leverage.  Platform.sh also exposes additional information to your application that way, including information like database credentials, the host or port it can use, and so forth.
 
 There are four different types of variable: 
 
-* application-provided variables
-* user-provided project variables
-* user-provided environment variables
-* Platform.sh-provided environment variables
+* [application-provided variables](/development/variables.html#application-provided-variables)
+* [user-provided project variables](/development/variables.html#project-variables)
+* [user-provided environment variables](/development/variables.html#environment-variables)
+* [Platform.sh-provided environment variables](/development/variables.html#platformsh-provided-variables)
+
+All of those may be simple strings or base64-encoded JSON-serialized values.  In case of name collisions, Platform.sh-provided values override user-provided environment variables, which override user-provided project-level variables, which override application-provided variables.  (That is, lower items in the list above take precedence.)
+
+## Comparison of variable types
+
+| Type          | Definer     | Scope       | Inheritance | Build | Runtime  |
```

Comment:
Should the list above be removed in favor of just the table, and giving the table the links?  I feel like that would be less-redundant.